app/Models/Beatmapset.php

```diff
@@ -469,17 +469,21 @@ public function fetchBeatmapsetArchive()
             return false;
         }
 
-        $contents = file_get_contents($url);
-        if ($contents === false) {
-            throw new BeatmapProcessorException('Error retrieving beatmap');
-        }
+        $curl = curl_init($url);
```

Comment:
does this need to handle redirects? (301/302)

Reply:
I don't believe so. But it's just a curl option that can't break anything, yeah?